File: src/FoodLoop.Domain/Entities/Organization.cs
using FoodLoop.Domain.Common;
using FoodLoop.Domain.Enums;
namespace FoodLoop.Domain.Entities;
public sealed class Organization : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string LicenseNumber { get; set; } = string.Empty;
    public OrganizationType Type { get; set; }
    public OrganizationStatus Status { get; set; } = OrganizationStatus.Pending;
    public string Address { get; set; } = string.Empty;
    public byte[] RowVersion { get; set; } = [];
}

File: frontend/src/types/organization.ts
// Organization enums, exactly as the backend serializes them (FoodLoop.Domain.Enums).
import type { OrganizationStatus, OrganizationType } from '../lib/api/auth'

export type { OrganizationStatus, OrganizationType }

export const ORGANIZATION_STATUSES = ['Active', 'Pending', 'Suspended', 'Rejected'] as const satisfies readonly OrganizationStatus[]

export const ORGANIZATION_TYPE_LABELS: Record<OrganizationType, string> = {
  Donor: 'Donor',
  Beneficiary: 'Beneficiary',
}
